src/Database/LSMTree/Internal/RunReaders.hs

```diff
@@ -104,35 +102,14 @@ data Reader m fhandle =
 
 type KOp m fhandle = (SerialisedKey, Entry SerialisedValue (BlobRef m fhandle))
 
--- | On equal keys, elements from runs earlier in the list are yielded first.
--- This means that the list of runs should be sorted from new to old.
 new :: forall h .
      HasFS IO h
   -> HasBlockIO IO h
-  -> Maybe WB.WriteBuffer
-  -> [Run IO (FS.Handle h)]
-  -> IO (Maybe (Readers IO (FS.Handle h)))
-new fs hbio = newAtOffsetMaybe fs hbio Nothing
-
--- | On equal keys, elements from runs earlier in the list are yielded first.
--- This means that the list of runs should be sorted from new to old.
-newAtOffset :: forall h .
-     HasFS IO h
-  -> HasBlockIO IO h
-  -> SerialisedKey  -- ^ offset
-  -> Maybe WB.WriteBuffer
-  -> [Run IO (FS.Handle h)]
-  -> IO (Maybe (Readers IO (FS.Handle h)))
-newAtOffset fs hbio offset = newAtOffsetMaybe fs hbio (Just offset)
-
-newAtOffsetMaybe :: forall h .
-     HasFS IO h
-  -> HasBlockIO IO h
   -> Maybe SerialisedKey  -- ^ offset
   -> Maybe WB.WriteBuffer
   -> [Run IO (FS.Handle h)]
   -> IO (Maybe (Readers IO (FS.Handle h)))
-newAtOffsetMaybe fs hbio mOffset wbs runs = do
+new fs hbio mOffset wbs runs = do
```

Comment:
Add the TODO here to convert the list of runs to a vector of runs, to avoid conversions.